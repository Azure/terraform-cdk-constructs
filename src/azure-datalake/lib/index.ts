export * from "./filesystem";
export * from "./path";
export * from "./datalake";
