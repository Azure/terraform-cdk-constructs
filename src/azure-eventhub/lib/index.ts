export * from "./authorization";
export * from "./eventhubcluster";
export * from "./consumer";
export * from "./instance";
export * from "./kusto-connection";
export * from "./namespace";
