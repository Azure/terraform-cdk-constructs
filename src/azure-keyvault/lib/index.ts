export * from './certificate'
export * from './key'
export * from './policy'
export * from './secret'
export * from './vault'

