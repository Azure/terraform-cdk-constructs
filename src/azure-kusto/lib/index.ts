export * from "./cluster";
export * from "./compute-specification";
export * from "./database";
