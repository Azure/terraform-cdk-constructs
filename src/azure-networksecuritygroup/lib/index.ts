export * from './preconfigured-rules';
export * from './securitygroup';
