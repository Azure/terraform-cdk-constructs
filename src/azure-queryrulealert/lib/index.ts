export * from './query-rule-alert'