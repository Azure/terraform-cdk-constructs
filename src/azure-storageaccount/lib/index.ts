export * from './account';
export * from './container';
export * from './fileshare';
export * from './queue';
export * from './table';
