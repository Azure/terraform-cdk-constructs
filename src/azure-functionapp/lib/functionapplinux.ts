import {
  LinuxFunctionApp,
  LinuxFunctionAppSiteConfig,
  LinuxFunctionAppSiteConfigApplicationStack,
  LinuxFunctionAppAuthSettings,
  LinuxFunctionAppAuthSettingsV2,
  LinuxFunctionAppIdentity,
  LinuxFunctionAppConnectionString,
} from "@cdktf/provider-azurerm/lib/linux-function-app";
import { ResourceGroup } from "@cdktf/provider-azurerm/lib/resource-group";
import { ServicePlan } from "@cdktf/provider-azurerm/lib/service-plan";
import { StorageAccount } from "@cdktf/provider-azurerm/lib/storage-account";
import { Construct } from "constructs";
import { AzureResource } from "../../core-azure/lib";

/**
 * Properties for the Azure Linux Function App
 */
export interface FunctionAppLinuxProps {
  /**
   * The name of the Function App. This name must be unique within Azure.
   */
  readonly name: string;

  /**
   * The Azure Region where the Function App will be deployed, e.g., 'East US', 'West Europe'.
   */
  readonly location: string;

  /**
   * An optional reference to the resource group in which to deploy the Function App.
   * If not provided, the Function App will be deployed in the default resource group.
   */
  readonly resourceGroup?: ResourceGroup;

  /**
   * An optional reference to the storage account to be used by the Function App.
   * If not provided, a new storage account will be created.
   */
  readonly storageAccount?: StorageAccount;

  /**
   * Optional runtime version specification for the Function App, such as Node.js, .NET, or Java version.
   */
  readonly runtimeVersion?: LinuxFunctionAppSiteConfigApplicationStack;

  /**
   * Optional ID of an existing App Service Plan to host the Function App.
   * If not provided, a new plan will be created.
   */
  readonly servicePlan?: ServicePlan;

  /**
   * Optional SKU for the App Service Plan, defines the pricing tier and capabilities.
   */
  readonly servicePlanSku?: string;

  /**
   * Optional tags for categorizing and managing the Function App resources within Azure.
   */
  readonly tags?: { [key: string]: string };

  /**
   * Optional site configuration for additional settings like environment variables, and connection strings.
   */
  readonly siteConfig?: LinuxFunctionAppSiteConfig;

  /**
   * Optional ID for the App Service Environment to be used by the service plan.
   */
  readonly servicePlanAppServiceEnvironmentId?: string;

  /**
   * Optional maximum count of elastic workers for the App Service Plan.
   */
  readonly servicePlanMaximumElasticWorkerCount?: number;

  /**
   * Optional worker count for the App Service Plan.
   */
  readonly servicePlanWorkerCount?: number;

  /**
   * Optional flag to enable per-site scaling for the App Service Plan.
   */
  readonly servicePlanPerSiteScalingEnabled?: boolean;

  /**
   * Optional flag to enable zone balancing for the App Service Plan.
   */
  readonly servicePlanZoneBalancingEnabled?: boolean;

  /**
   * Optional authentication settings for the Function App.
   */
  readonly authSettings?: LinuxFunctionAppAuthSettings;

  /**
   * Optional advanced version of authentication settings for the Function App.
   */
  readonly authSettingsV2?: LinuxFunctionAppAuthSettingsV2;

  /**
   * Optional identity configuration for the Function App, for use in Managed Service Identity scenarios.
   */
  readonly identity?: LinuxFunctionAppIdentity;

  /**
   * Optional flag to enforce HTTPS only traffic.
   */
  readonly httpsOnly?: boolean;

  /**
   * Optional flag to enable client certificate authentication.
   */
  readonly clientCertificateEnabled?: boolean;

  /**
   * Optional mode for client certificate requirement (e.g., 'Required', 'Optional').
   */
  readonly clientCertificateMode?: string;

  /**
   * Optional paths that are excluded from client certificate authentication.
   */
  readonly clientCertificateExclusionPaths?: string;

  /**
   * Optional version setting for the Azure Functions runtime.
   */
  readonly functionsExtensionVersion?: string;

  /**
   * Optional flag to enable or disable the Function App.
   */
  readonly enabled?: boolean;

  /**
   * Optional flag to enable built-in logging capabilities.
   */
  readonly builtinLoggingEnabled?: boolean;

  /**
   * Optional connection string for external services or databases.
   */
  readonly connectionString?: LinuxFunctionAppConnectionString[];

  /**
   * Optional flag to enable or disable public network access to the Function App.
   */
  readonly publicNetworkAccessEnabled?: boolean;

  /**
   * Optional flag indicating if the storage account uses a Managed Identity.
   */
  readonly storageUsesManagedIdentity?: boolean;

  /**
   * Optional access key for the storage account.
   */
  readonly storageAccountAccessKey?: string;

  /**
   * Optional ID of a virtual network subnet for the Function App.
   */
  readonly virtualNetworkSubnetId?: string;

  /**
   * Optional path to a ZIP file for deployment to the Function App.
   */
  readonly zipDeployFile?: string;

  /**
   * Application settings for the Azure Function App.
   *
   * @property { [key: string]: string } appSettings - A collection of key-value pairs that contain the settings.
   *
   * Note on Runtime Settings:
   * - 'node_version' in 'site_config' sets the Node.js version.
   *   Terraform assigns this value to 'WEBSITE_NODE_DEFAULT_VERSION' in app settings.
   * - 'functions_extension_version' sets the Azure Functions runtime version.
   *   Terraform assigns this value to 'FUNCTIONS_EXTENSION_VERSION' in app settings.
   *
   * Note on Storage Settings:
   * - Properties like 'storage_account_access_key' are used for storage configurations.
   *   Terraform assigns these values to keys like 'WEBSITE_CONTENTAZUREFILECONNECTIONSTRING',
   *   'AzureWebJobsStorage' in app settings.
   *
   * Note on Application Insights Settings:
   * - Use 'application_insights_connection_string' and 'application_insights_key' for Application Insights configurations.
   *   Terraform assigns these to 'APPINSIGHTS_INSTRUMENTATIONKEY' and 'APPLICATIONINSIGHTS_CONNECTION_STRING' in app settings.
   *
   * Note on Health Check Settings:
   * - 'health_check_eviction_time_in_min' configures health check settings.
   *   Terraform assigns this value to 'WEBSITE_HEALTHCHECK_MAXPINGFAILURES' in app settings.
   *
   * Note on Storage Account Restriction:
   * - To restrict your storage account to a virtual network, set 'WEBSITE_CONTENTOVERVNET' to 1 in app settings.
   *   Ensure a predefined share is created for this configuration.
   */
  readonly appSettings?: { [key: string]: string };
}

export class FunctionAppLinux extends AzureResource {
  public id: string;
  public readonly defaultHostname: string;
  public readonly kind: string;
  public readonly name: string;
  public readonly storageAccount: StorageAccount;
  public readonly servicePlan: ServicePlan;
  public resourceGroup: ResourceGroup;
  /**
   * Constructs a new FunctionAppLinux.
   *
   * @param scope - The scope in which to define this construct.
   * @param id - The ID of this construct.
   * @param props - The properties for configuring the Azure Function App on Linux. The properties include:
   *                - `name`: Required. Unique name for the Function App within Azure.
   *                - `location`: Required. Azure Region for deployment.
   *                - `resourceGroup`: Optional. Reference to the resource group for deployment.
   *                - `storageAccount`: Optional. Reference to the storage account used by the Function App.
   *                - `runtimeVersion`: Optional. Specifies the runtime version (Node.js, .NET, Java, etc.).
   *                - `servicePlan`: Optional. ID of an existing App Service Plan.
   *                - `servicePlanSku`: Optional. SKU for the App Service Plan.
   *                - `tags`: Optional. Tags for resource management.
   *                - `siteConfig`: Optional. Additional site configuration settings.
   *                - Additional optional properties as described in `FunctionAppLinuxProps` interface.
   *
   * Example usage:
   * ```typescript
   * new FunctionAppLinux(this, 'premiumFA', {
   *   name: `faprem${this.name}`,
   *   location: 'eastus',
   *   servicePlanSku: ServicePlanSkus.PremiumEP1,
   *   runtimeVersion: {
   *     dotnetVersion: '5.0',
   *   },
   *   tags: {
   *     "test": "test"
   *   }
   * });
   * ```
   */
  constructor(scope: Construct, id: string, props: FunctionAppLinuxProps) {
    super(scope, id);

    this.resourceGroup = this.setupResourceGroup(props);
    this.storageAccount = this.setupStorageAccount(props);
    this.servicePlan = this.setupServicePlan(props);

    this.resourceGroup = this.resourceGroup;

    // Default siteConfig settings for each hosting plan
    let defaultSiteConfig: LinuxFunctionAppSiteConfig = {
      use32BitWorker: false,
      applicationStack: {
        nodeVersion: "14",
      },
    };

    // Merge user provided siteConfig with defaults
    const siteConfig = { ...defaultSiteConfig, ...props.siteConfig };

    const defaults = {
      httpsOnly: props.httpsOnly || true,
      clientCertificateEnabled: props.clientCertificateEnabled || false,
      clientCertificateMode: props.clientCertificateMode || "Required",
      identity: props.identity || { type: "SystemAssigned" },
      clientCertificateExclusionPaths:
        props.clientCertificateExclusionPaths || "",
      functionsExtensionVersion: props.functionsExtensionVersion || "~4",
      enabled: props.enabled || true,
      builtinLoggingEnabled: props.builtinLoggingEnabled || true,
      publicNetworkAccessEnabled: props.publicNetworkAccessEnabled || true,
      storageUsesManagedIdentity: props.storageUsesManagedIdentity || true,
      storageAccountAccessKey: undefined || props.storageAccountAccessKey,
    };

    // Create the Function App
    const functionApp = new LinuxFunctionApp(this, "FunctionApp", {
      ...defaults,
      name: props.name,
      location: props.location,
      resourceGroupName: this.resourceGroup.name,
      servicePlanId: this.servicePlan.id,
      storageAccountName: this.storageAccount.name,
      tags: props.tags,
      siteConfig: siteConfig,
      appSettings: props.appSettings,
      authSettings: props.authSettings,
      authSettingsV2: props.authSettingsV2,
      connectionString: props.connectionString,
      virtualNetworkSubnetId: props.virtualNetworkSubnetId,
      zipDeployFile: props.zipDeployFile,
    });

    this.id = functionApp.id;
    this.defaultHostname = functionApp.defaultHostname;
    this.kind = functionApp.kind;
    this.name = functionApp.name;
  }

  private setupResourceGroup(props: FunctionAppLinuxProps): ResourceGroup {
    if (!props.resourceGroup) {
      // Create a new resource group
      const newResourceGroup = new ResourceGroup(this, "rg", {
        name: `rg-${props.name}`,
        location: props.location,
        tags: props.tags,
      });
      // Use the name of the new resource group
      return newResourceGroup;
    } else {
      // Use the provided resource group name
      return props.resourceGroup;
    }
  }

  private setupStorageAccount(props: FunctionAppLinuxProps): StorageAccount {
    // Reference or create a new storage account that will be used by the Function App to store files
    if (!props.storageAccount) {
      return new StorageAccount(this, "sa", {
        name: `${props.name}stacc`,
        resourceGroupName: this.resourceGroup.name,
        location: props.location,
        accountReplicationType: "LRS",
        accountTier: "Standard",
        minTlsVersion: "TLS1_2",
        tags: props.tags,
        publicNetworkAccessEnabled: false,
        networkRules: {
          bypass: ["AzureServices"],
          defaultAction: "Deny",
        },
      });
    } else {
      return props.storageAccount;
    }
  }

  private setupServicePlan(props: FunctionAppLinuxProps): ServicePlan {
    const servicePlanSKU = props.servicePlanSku || "Y1";

    // Use an existing App Service Plan if the ID is provided, otherwise create a new one
    if (!props.servicePlan) {
      return new ServicePlan(this, "AppServicePlan", {
        name: `${props.name}asp`,
        location: props.location,
        resourceGroupName: this.resourceGroup.name,
        skuName: servicePlanSKU,
        osType: "Linux",
        appServiceEnvironmentId: props.servicePlanAppServiceEnvironmentId,
        maximumElasticWorkerCount: props.servicePlanMaximumElasticWorkerCount,
        workerCount: props.servicePlanWorkerCount,
        perSiteScalingEnabled: props.servicePlanPerSiteScalingEnabled,
        zoneBalancingEnabled: props.servicePlanZoneBalancingEnabled,
        tags: props.tags,
      });
    } else {
      return props.servicePlan;
    }
  }
}
