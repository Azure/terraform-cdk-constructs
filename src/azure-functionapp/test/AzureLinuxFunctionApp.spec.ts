import { AzurermProvider } from "@cdktf/provider-azurerm/lib/provider";
import { Testing, TerraformStack } from "cdktf";
import "cdktf/lib/testing/adapters/jest";
import { exampleAzureLinuxFunctionApp } from "./ExampleAzureLinuxFunctionApp";
import * as func from "..";
import * as util from "../../util/azureTenantIdHelpers";

describe("Azure Linux Function App With Defaults", () => {
  let stack: TerraformStack;
  let fullSynthResult: any;

  beforeEach(() => {
    const app = Testing.app();
    stack = new TerraformStack(app, "test");

    new AzurermProvider(stack, "azureFeature", { features: {} });

    new func.FunctionAppLinux(stack, "testAzureLinuxFunctionApp", {
      name: "fatest",
      location: "eastus",
    });

    fullSynthResult = Testing.fullSynth(stack); // Save the result for reuse
  });

  it("renders an Azure Linux Function App with defaults and checks snapshot", () => {
    expect(Testing.synth(stack)).toMatchSnapshot(); // Compare the already prepared stack
  });

  it("check if the produced terraform configuration is valid", () => {
    expect(fullSynthResult).toBeValidTerraform(); // Use the saved result
  });

  it("check if this can be planned", () => {
    expect(fullSynthResult).toPlanSuccessfully(); // Use the saved result
  });
});

describe("Linux Function App Example", () => {
  it("renders the Linux Function App and checks snapshot", () => {
    // Need to remove the tenant_id from the snapshot as it will change wherever the test is run
    const output = Testing.synth(
      new exampleAzureLinuxFunctionApp(
        Testing.app(),
        "testAzureLinuxFunctionApp",
      ),
    );
    const myObject: Record<string, any> = JSON.parse(output);

    expect(util.removeTenantIdFromSnapshot(myObject)).toMatchSnapshot();
  });

  it("check if the produced terraform configuration is valid", () => {
    // We need to do a full synth to plan the terraform configuration
    expect(
      Testing.fullSynth(
        new exampleAzureLinuxFunctionApp(
          Testing.app(),
          "testAzureLinuxFunctionApp",
        ),
      ),
    ).toBeValidTerraform();
  });

  it("check if this can be planned", () => {
    // We need to do a full synth to plan the terraform configuration
    expect(
      Testing.fullSynth(
        new exampleAzureLinuxFunctionApp(
          Testing.app(),
          "testAzureLinuxFunctionApp",
        ),
      ),
    ).toPlanSuccessfully();
  });
});
