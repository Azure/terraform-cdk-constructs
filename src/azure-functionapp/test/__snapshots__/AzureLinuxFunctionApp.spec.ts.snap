// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Azure Linux Function App With Defaults renders an Azure Linux Function App with defaults and checks snapshot 1`] = `
"{
  "provider": {
    "azurerm": [
      {
        "features": {
        }
      }
    ]
  },
  "resource": {
    "azurerm_linux_function_app": {
      "testAzureLinuxFunctionApp_2B86F984": {
        "builtin_logging_enabled": true,
        "client_certificate_enabled": false,
        "client_certificate_exclusion_paths": "",
        "client_certificate_mode": "Required",
        "enabled": true,
        "functions_extension_version": "~4",
        "https_only": true,
        "identity": {
          "type": "SystemAssigned"
        },
        "location": "eastus",
        "name": "fatest",
        "public_network_access_enabled": true,
        "resource_group_name": "\${azurerm_resource_group.testAzureLinuxFunctionApp_rg_9D21757F.name}",
        "service_plan_id": "\${azurerm_service_plan.testAzureLinuxFunctionApp_AppServicePlan_41C3FFFD.id}",
        "site_config": {
          "application_stack": {
            "node_version": "14"
          },
          "use_32_bit_worker": false
        },
        "storage_account_name": "\${azurerm_storage_account.testAzureLinuxFunctionApp_sa_4C0B969D.name}",
        "storage_uses_managed_identity": true
      }
    },
    "azurerm_resource_group": {
      "testAzureLinuxFunctionApp_rg_9D21757F": {
        "location": "eastus",
        "name": "rg-fatest"
      }
    },
    "azurerm_service_plan": {
      "testAzureLinuxFunctionApp_AppServicePlan_41C3FFFD": {
        "location": "eastus",
        "name": "fatestasp",
        "os_type": "Linux",
        "resource_group_name": "\${azurerm_resource_group.testAzureLinuxFunctionApp_rg_9D21757F.name}",
        "sku_name": "Y1"
      }
    },
    "azurerm_storage_account": {
      "testAzureLinuxFunctionApp_sa_4C0B969D": {
        "account_replication_type": "LRS",
        "account_tier": "Standard",
        "location": "eastus",
        "min_tls_version": "TLS1_2",
        "name": "fateststacc",
        "network_rules": {
          "bypass": [
            "AzureServices"
          ],
          "default_action": "Deny"
        },
        "public_network_access_enabled": false,
        "resource_group_name": "\${azurerm_resource_group.testAzureLinuxFunctionApp_rg_9D21757F.name}"
      }
    }
  },
  "terraform": {
    "required_providers": {
      "azurerm": {
        "source": "azurerm",
        "version": "3.83.0"
      }
    }
  }
}"
`;

exports[`Linux Function App Example renders the Linux Function App and checks snapshot 1`] = `
{
  "data": {
    "azurerm_monitor_diagnostic_categories": {
      "consumptionFA_diagsettings_diagcategories_AF51DBB0": {
        "resource_id": "\${azurerm_linux_function_app.consumptionFA_FunctionApp_0D946802.id}",
      },
    },
  },
  "output": {
    "default_hostname": {
      "value": "\${azurerm_linux_function_app.consumptionFA_FunctionApp_0D946802.default_hostname}",
    },
    "function_app_id": {
      "value": "\${azurerm_linux_function_app.consumptionFA_FunctionApp_0D946802.id}",
    },
    "function_app_kind": {
      "value": "\${azurerm_linux_function_app.consumptionFA_FunctionApp_0D946802.kind}",
    },
    "function_app_name": {
      "value": "\${azurerm_linux_function_app.consumptionFA_FunctionApp_0D946802.name}",
    },
    "resource_group_name": {
      "value": "\${azurerm_resource_group.rg.name}",
    },
  },
  "provider": {
    "azurerm": [
      {
        "features": {
          "resource_group": {
            "prevent_deletion_if_contains_resources": false,
          },
        },
      },
    ],
  },
  "resource": {
    "azurerm_linux_function_app": {
      "consumptionFA2_FunctionApp_F726133A": {
        "builtin_logging_enabled": true,
        "client_certificate_enabled": false,
        "client_certificate_exclusion_paths": "",
        "client_certificate_mode": "Required",
        "enabled": true,
        "functions_extension_version": "~4",
        "https_only": true,
        "identity": {
          "type": "SystemAssigned",
        },
        "location": "eastus",
        "name": "fa\${var.name}2",
        "public_network_access_enabled": true,
        "resource_group_name": "\${azurerm_resource_group.rg.name}",
        "service_plan_id": "\${azurerm_service_plan.consumptionFA_AppServicePlan_8DD812C2.id}",
        "site_config": {
          "application_stack": {
            "node_version": "14",
          },
          "cors": {
            "allowed_origins": [
              "*",
            ],
          },
          "use_32_bit_worker": false,
        },
        "storage_account_name": "\${azurerm_storage_account.consumptionFA_sa_CFEEEED8.name}",
        "storage_uses_managed_identity": true,
        "tags": {
          "test": "test",
        },
      },
      "consumptionFA_FunctionApp_0D946802": {
        "builtin_logging_enabled": true,
        "client_certificate_enabled": false,
        "client_certificate_exclusion_paths": "",
        "client_certificate_mode": "Required",
        "enabled": true,
        "functions_extension_version": "~4",
        "https_only": true,
        "identity": {
          "type": "SystemAssigned",
        },
        "location": "eastus",
        "name": "fa\${var.name}",
        "public_network_access_enabled": true,
        "resource_group_name": "\${azurerm_resource_group.rg.name}",
        "service_plan_id": "\${azurerm_service_plan.consumptionFA_AppServicePlan_8DD812C2.id}",
        "site_config": {
          "application_stack": {
            "node_version": "14",
          },
          "use_32_bit_worker": false,
        },
        "storage_account_name": "\${azurerm_storage_account.consumptionFA_sa_CFEEEED8.name}",
        "storage_uses_managed_identity": true,
        "tags": {
          "test": "test",
        },
      },
      "premiumFA_FunctionApp_BCBE916E": {
        "builtin_logging_enabled": true,
        "client_certificate_enabled": false,
        "client_certificate_exclusion_paths": "",
        "client_certificate_mode": "Required",
        "enabled": true,
        "functions_extension_version": "~4",
        "https_only": true,
        "identity": {
          "type": "SystemAssigned",
        },
        "location": "eastus",
        "name": "faprem\${var.name}",
        "public_network_access_enabled": true,
        "resource_group_name": "\${azurerm_resource_group.premiumFA_rg_5BA9B2FE.name}",
        "service_plan_id": "\${azurerm_service_plan.premiumFA_AppServicePlan_8A862E92.id}",
        "site_config": {
          "application_stack": {
            "node_version": "14",
          },
          "use_32_bit_worker": false,
        },
        "storage_account_name": "\${azurerm_storage_account.premiumFA_sa_DC65AFA0.name}",
        "storage_uses_managed_identity": true,
        "tags": {
          "test": "test",
        },
      },
      "servicePlanFA_FunctionApp_897DB6CB": {
        "builtin_logging_enabled": true,
        "client_certificate_enabled": false,
        "client_certificate_exclusion_paths": "",
        "client_certificate_mode": "Required",
        "enabled": true,
        "functions_extension_version": "~4",
        "https_only": true,
        "identity": {
          "type": "SystemAssigned",
        },
        "location": "eastus",
        "name": "fasp\${var.name}",
        "public_network_access_enabled": true,
        "resource_group_name": "\${azurerm_resource_group.servicePlanFA_rg_758B3FD5.name}",
        "service_plan_id": "\${azurerm_service_plan.servicePlanFA_AppServicePlan_6B09BCCF.id}",
        "site_config": {
          "application_stack": {
            "node_version": "14",
          },
          "cors": {
            "allowed_origins": [
              "*",
            ],
          },
          "use_32_bit_worker": false,
        },
        "storage_account_name": "\${azurerm_storage_account.servicePlanFA_sa_084D89FA.name}",
        "storage_uses_managed_identity": true,
        "tags": {
          "test": "test",
        },
      },
    },
    "azurerm_log_analytics_workspace": {
      "log_analytics": {
        "location": "eastus",
        "name": "la-\${var.name}",
        "resource_group_name": "\${azurerm_resource_group.rg.name}",
      },
    },
    "azurerm_monitor_diagnostic_setting": {
      "consumptionFA_diagsettings_ABB8EE42": {
        "dynamic": {
          "enabled_log": {
            "content": {
              "category": "\${enabled_log.value}",
            },
            "for_each": "\${tolist(data.azurerm_monitor_diagnostic_categories.consumptionFA_diagsettings_diagcategories_AF51DBB0.log_category_types)}",
          },
          "metric": {
            "content": {
              "category": "\${metric.value}",
            },
            "for_each": "\${tolist(data.azurerm_monitor_diagnostic_categories.consumptionFA_diagsettings_diagcategories_AF51DBB0.metrics)}",
          },
        },
        "log_analytics_workspace_id": "\${azurerm_log_analytics_workspace.log_analytics.id}",
        "name": "diagsettings",
        "target_resource_id": "\${azurerm_linux_function_app.consumptionFA_FunctionApp_0D946802.id}",
      },
    },
    "azurerm_resource_group": {
      "premiumFA_rg_5BA9B2FE": {
        "location": "eastus",
        "name": "rg-faprem\${var.name}",
        "tags": {
          "test": "test",
        },
      },
      "rg": {
        "location": "eastus",
        "name": "rg-\${var.name}",
        "tags": {
          "test": "test",
        },
      },
      "servicePlanFA_rg_758B3FD5": {
        "location": "eastus",
        "name": "rg-fasp\${var.name}",
        "tags": {
          "test": "test",
        },
      },
    },
    "azurerm_service_plan": {
      "consumptionFA_AppServicePlan_8DD812C2": {
        "location": "eastus",
        "name": "fa\${var.name}asp",
        "os_type": "Linux",
        "resource_group_name": "\${azurerm_resource_group.rg.name}",
        "sku_name": "Y1",
        "tags": {
          "test": "test",
        },
      },
      "premiumFA_AppServicePlan_8A862E92": {
        "location": "eastus",
        "name": "faprem\${var.name}asp",
        "os_type": "Linux",
        "resource_group_name": "\${azurerm_resource_group.premiumFA_rg_5BA9B2FE.name}",
        "sku_name": "EP1",
        "tags": {
          "test": "test",
        },
      },
      "servicePlanFA_AppServicePlan_6B09BCCF": {
        "location": "eastus",
        "name": "fasp\${var.name}asp",
        "os_type": "Linux",
        "resource_group_name": "\${azurerm_resource_group.servicePlanFA_rg_758B3FD5.name}",
        "sku_name": "B1",
        "tags": {
          "test": "test",
        },
      },
    },
    "azurerm_storage_account": {
      "consumptionFA_sa_CFEEEED8": {
        "account_replication_type": "LRS",
        "account_tier": "Standard",
        "location": "eastus",
        "min_tls_version": "TLS1_2",
        "name": "fa\${var.name}stacc",
        "network_rules": {
          "bypass": [
            "AzureServices",
          ],
          "default_action": "Deny",
        },
        "public_network_access_enabled": false,
        "resource_group_name": "\${azurerm_resource_group.rg.name}",
        "tags": {
          "test": "test",
        },
      },
      "premiumFA_sa_DC65AFA0": {
        "account_replication_type": "LRS",
        "account_tier": "Standard",
        "location": "eastus",
        "min_tls_version": "TLS1_2",
        "name": "faprem\${var.name}stacc",
        "network_rules": {
          "bypass": [
            "AzureServices",
          ],
          "default_action": "Deny",
        },
        "public_network_access_enabled": false,
        "resource_group_name": "\${azurerm_resource_group.premiumFA_rg_5BA9B2FE.name}",
        "tags": {
          "test": "test",
        },
      },
      "servicePlanFA_sa_084D89FA": {
        "account_replication_type": "LRS",
        "account_tier": "Standard",
        "location": "eastus",
        "min_tls_version": "TLS1_2",
        "name": "fasp\${var.name}stacc",
        "network_rules": {
          "bypass": [
            "AzureServices",
          ],
          "default_action": "Deny",
        },
        "public_network_access_enabled": false,
        "resource_group_name": "\${azurerm_resource_group.servicePlanFA_rg_758B3FD5.name}",
        "tags": {
          "test": "test",
        },
      },
    },
  },
  "terraform": {
    "required_providers": {
      "azurerm": {
        "source": "azurerm",
        "version": "3.83.0",
      },
    },
  },
  "variable": {
    "name": {
      "default": "test",
      "description": "System name used to randomize the resources",
      "type": "string",
    },
  },
}
`;
